target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/dsa.c

```diff
@@ -2013,12 +2013,42 @@ static int rtl83xx_port_pre_bridge_flags(struct dsa_switch *ds, int port, struct
 		features |= BR_MCAST_FLOOD;
 	if (priv->r->enable_bcast_flood)
 		features |= BR_BCAST_FLOOD;
+	if (priv->family_id == RTL9300_FAMILY_ID ||
+			priv->family_id == RTL9310_FAMILY_ID)
+		features |= BR_ISOLATED;
+
 	if (flags.mask & ~(features))
 		return -EINVAL;
 
 	return 0;
 }
 
+static void rtl93xx_port_isolation(struct rtl838x_switch_priv *priv, int port, bool isolate)
```

Comment:
no, this was not what was requested. Now you just removed the function pointer and still added something which is only for rtl93xx. So just moved the function and broke the code in the function body.